utils/util_math.py

```diff
@@ -0,0 +1,31 @@
+from math import sqrt
```

Comment:
The sole purpose of this file was to avoid adding an unnecessary dependency on numpy, but the memory analysis already requires it, so you can safely delete this and use the numpy methods where needed. As a side note, `sqrt = lambda x: x ** 0.5`.